pcap/pcap.h

```diff
@@ -352,6 +352,7 @@ PCAP_API const char *pcap_tstamp_type_val_to_description(int);
 
 #ifdef __linux__
 PCAP_API int	pcap_set_protocol_linux(pcap_t *, int);
+PCAP_API int	pcap_set_fanout_linux(pcap_t *, uint16_t, uint16_t);
```

Comment:
1. You've commented on the previous version of this PR (not that it really matters, the diff is minor).
2. This line is inside `#ifdef __linux__` (just like `pcap_set_protocol_linux()`). It will only get built for Linux, where it's relevant.
Or, have I missed a problem you're pointing to?

Reply:
1. Oh, sorry.
2. I see. I was aiming more at not breaking builds on other platforms even when used. But it's arguable that a system that expects fanout will have a design depending on it, so maybe breaking at build time is better. In any case, I think a nice message when the build breaks may be more user friendly than a link time missing symbol.

Reply:
* As far as I can see, the build of libpcap won't fail due to missing symbols.
* The failure will be witnessed during runtime of a client application calling `pcap_set_fanout_linux()` when `PACKET_FANOUT` wasn't defined. In such case, a descriptive error would be returned.
* The idea is that a function named `*_linux()` will be available on the Linux platform regardless if the inner implementation can be executed or not.
* Despite the above, I'm not "married" to this solution. We can hide `pcap_set_fanout_linux()` within `#ifdef PACKET_FANOUT`. I feel comfortable with that option as well.
* I have no strong opinion about both options.

Reply:
1. I didn't mean libpcap's build, but a user's application.
2. Yes, that's OK. I'm thinking of, say, a FreeBSD user that mistakenly tries to build a program which depends on this (and, thus, on Linux), will the error *they* receive be clear enough?
3. I agree. It shouldn't *work* or be used on other platforms, but if it's clear to the user why the program doesn't work, then that's better.
4. I'm happier with the current solution than with `#ifdef`ing the whole function, mostly because it's easier to tell the users your program runs only on Linux than it is to tell them it runs on kernels above a given version, but might run in this and the other distros because they backport stuff and the like. However, as I said before, an application that depends on having fanout is probably an all or nothing situation, so my point here is rather moot.

Reply:
Note using undeclared functions isn't itself an error (most compilers do issue warnings and allow turning them into errors, but that may be ignored), so a program written for Linux using this interface may build but fail to run.
Although now that I'm thinking more about that, I'm unsure there's a portable way to inform of that error to the downstream user/programmer.

Reply:
>     2\. Yes, that's OK. I'm thinking of, say, a FreeBSD user that mistakenly tries to build a program which depends on this (and, thus, on Linux), will the error _they_ receive be clear enough?

> Note using undeclared functions isn't itself an error (most compilers do issue warnings and allow turning them into errors, but that may be ignored), so a program written for Linux using this interface may build but fail to run.

Answer is similar:
1. Same behaviour like `pcap_set_protocol_linux` (another Linux only API which already exist in the code base).
2. Users using "unsafe" compiler flags (or more precisely, failing to add proper compiler flags)... I wish them best of luck with their code in so many ways...
3. I can start describing the various errors users may get on different points of the compilation/linkage/execution process in case the user decides to be creative in ignoring/working around compilation warnings, but I really do think it is meaningless (even the function has `linux` in its name, that should mean something, no?).

Reply:
OK, that makes sense.